src/sentry_tracing.c

```diff
@@ -161,9 +161,9 @@ sentry__transaction_decref(sentry_transaction_t *tx)
 }
 
 sentry_span_t *
-sentry__span_new(sentry_value_t inner)
+sentry__span_new(sentry_transaction_t *tx, sentry_value_t inner)
```

Comment:
this change to the signature caused a series of cascading changes which are all internal, thankfully.

there should be no reason why a span could or should be started without some root transaction associated with it, so this was added in as a parameter to the "constructor".

setting the transaction after the creation of the span also caused some pointer-related issues, so this also resolves a bug.